Revert #6. Do not lose previously collected data

basilevs opened the pull request:
https://github.com/jenkinsci/rich-text-publisher-plugin/pull/6 has fundamentally changed the behavior, breaking existing pipelines. See https://github.com/jenkinsci/rich-text-publisher-plugin/pull/6#issuecomment-1637511971 for details. This is a direct revert of the change, restoring the original accumulation behavior.

masterhard committed:
Merge pull request #6 from xjjx/master

use addOrReplaceAction

masterhard committed:
Upgraded pom file to fresh Jenkins Maven plugin. Deleted an incomplete test.
Upgraded to a fresh WikiText parser library.

masterhard committed:
[maven-release-plugin] prepare release rich-text-publisher-plugin-1.5

masterhard committed:
[maven-release-plugin] prepare for next development iteration

basilevs committed:
Revert #6. Do not lose previously collected data

https://github.com/jenkinsci/rich-text-publisher-plugin/pull/6 has fundamentally changed the behavior, breaking existing pipelines.
See https://github.com/jenkinsci/rich-text-publisher-plugin/pull/6#issuecomment-1637511971 for details.
This is a direct revert of the change, restoring the original accumulation behavior.

nneul commented:
I would suggest that the right way to address this wouldn't be a new command/function, but could just be a boolean option to the 'rtp' pipeline step for 'replace: true' if you wanted the new behavior.

That would be a minimally invasive code change and easy to use.

For the generic step, that could be exposed as a configuration element on the step.

I agree completely with the rollback in the immediate term, but expect that implementing above would be pretty simple.

basilevs committed:
Merge remote-tracking branch 'jenkinsci/master' into patch-1

basilevs committed:
JENKINS-72140 Exlplicitly indicate backward incompatibility

See https://www.jenkins.io/doc/developer/plugin-development/mark-a-plugin-incompatible/

Version 1.5 was backward incompatible in a silent manner. To avoid further complications, make the revert of backward incompatible changes explicit.
